zerver/openapi/zulip.yaml

```diff
@@ -9225,6 +9225,10 @@ paths:
                           array will be empty if `enable_drafts_synchronization` is set to `false`.
                         items:
                           $ref: "#/components/schemas/Draft"
+                      emoji_data_url:
+                        type: string
+                        description: |
+                          The URL to a JSON file that describes which emoji names map to which emoji codes.
```

Comment:
Here's a draft incorporating the other details of this API we want to specify, based on the chat thread:

```suggestion
                          The URL to a JSON file that describes which emoji names map to which emoji codes, for all Unicode emoji this Zulip server accepts.

                          The data at the given URL is a JSON object with one property, `name_to_code`. The value of that property is a JSON object where each key is an [emoji name](/api/add-reaction#parameter-emoji_name) for an available Unicode emoji, and each value is the corresponding [emoji code](/api/add-reaction#parameter-emoji_code). A given emoji code may appear under multiple emoji names.

                          The data at this URL is immutable: the same URL will always represent the same JSON data. This means that if a client sees the value of `emoji_data_url` is a URL it has previously fetched, it can rely on the data it fetched then and avoid fetching it again. (The server accomplishes this guarantee by incorporating a hash of the data in the URL.)
```

(But do wrap the paragraphs -- that'd just be annoying to do here in the GitHub comment editor.)

I think ideally we'd use the nice OpenAPI syntax for describing the structure of the JSON object. Not sure how much fiddling that might require. But I think for now, with just the one property `name_to_code`, this prose version works well enough.

Reply:
Hmm. That is a problem for the mobile app, then -- we don't have an easy way to get our hands on the HTTP headers.

If that's the situation, then I think we should go back to having an `emoji_data_hash` property in the API too.

Reply:
(Subsequent thread on the new PR for just this part of the changes: https://github.com/zulip/zulip/pull/22476#discussion_r923967310 )